### PR TITLE
[DSDF] Add new object type DSFD (abapGit#6948)

### PR DESCRIPTION
**Pull request by fvalves**
Added directory for new object type DSFD (abapGit#6948)

### CODE REVIEW

**Comment by fvalves** [comment]
Hi @schneidermic0 , I've added the corrections analogous to the DRAS ones here as well.

File: file-formats/dsfd/type/zif_aff_dsfd_v1.intf.abap [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  TYPES:
+    "! <p class="shorttext">CDS Scalar Function Definition</p>
+    "! CDS scalar function definition
+    BEGIN OF ty_main,
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header         TYPE zif_aff_types_v1=>ty_header_60,
```
Comment (schneidermic0) [inline file-formats/dsfd/type/zif_aff_dsfd_v1.intf.abap:13]: If you don't support key user tooling for DSDF (yet), I would change the header as well.

```suggestion
      header         TYPE zif_aff_types_v1=>ty_header_60_cloud,
```

**Review by schneidermic0 — COMMENTED** [review]
In general, it looks good to me. Maybe, just the header needs to be changed if DSFD have not supported ABAP language version "keyUser", yet.

**Comment by schneidermic0** [comment]
Welcome as contributor, @fvalves 👍

**Pull request merged**